Add the new WeDo 2.0 extension

cwillisf opened the pull request:
Requires LLK/scratchr2#3371 for online support
- Support extensions whose names contain '.'
- Add the WeDo 2.0 extension to the Extension Library
- Update the name and icon for the WeDo 1.0 extension
- Alter `resetPlugin()` to support a bug fix in the offline editor

Commit:
Handle extension name containing '.'

Commit:
Add WeDo 2.0 extension to the list in ExtensionManager

Commit:
Load extensions using resetPlugin callback

This allows `resetPlugin()` to finish completely before
loading extensions, fixing a bug in the offline editor.

Commit:
Fix warnings

Commit:
Update WeDo icons for transparent background

Resolves LLK/scratch-device-manager#12

cpseager commented on src/extensions/ExtensionManager.as:
```diff
@@ -49,7 +49,9 @@ public class ExtensionManager {
-	static public const wedoExt:String = 'LEGO WeDo';
+	static public const picoBoardExt:String = 'PicoBoard';
+	static public const wedoExt:String = 'LEGO WeDo 1.0';
+	static public const wedo2Ext:String = 'LEGO WeDo 2.0';
```
@cwillisf
Won't renaming original WeDo cause opening issues when loading saved files that already have 

"extensionName": "LEGO WeDo"

in their JSON?

cwillisf replied:
...yes. Good point. I'll probably revert this line.

cwillisf replied:
...and by "probably" I mean "unless I find a better way" -- not "maybe I'll decide to break all those projects"! :flushed:

Thanks for taking a look!

Commit:
Rename original WeDo extension back to 'LEGO WeDo'

Changing the internal name would break projects which contain a
reference to this extension. If we really want it to show up as 'LEGO
WeDo 1.0' in the extension library we should add a way to separate the
display name of an extension from its internal name.

Commit:
Added support for extension display name

    -Can now specify displayName which can differ from name
    -displayName defaults to name if not specified
